### PR TITLE
Ultra dry tests

### PR DESCRIPTION
**Pull request by adriendelsalle**
Description
---

Create ultra dry tests to fasten CI.
Uses `Configuration` status after being loaded instead of `dry_run`/`--dry-run` that still requires fetching the repodata.json and running the solver.

### CODE REVIEW

**Comment by adriendelsalle** [comment]
@wolfv This PR allows to save about 10min on the CI but also masks some issues met when executing real operations.
For example, the recent issue about Windows long paths not supported in `cmd.exe` causing errors when installing noarch packages is not caught here.

Any recommendations on where to run normal/dry/ultra_dry tests ?

**Comment by wolfv** [comment]
I think everything about CLI parsing can be run in ultra-dry mode.

File: include/mamba/api/configuration.hpp [hunk 1]
```diff
@@ -18,6 +18,33 @@
...
 
 
+#ifdef ENABLE_CONTEXT_DEBUG_PRINT
+#define CONTEXT_DEBUGGING                                                                          \
+    if (Configuration::instance().at("print_context_only").value<bool>())                          \
+    {                                                                                              \
+        Context::instance().debug_print();                                                         \
+        exit(0);                                                                                   \
+    }
+#define CONFIG_DEBUGGING                                                                           \
```
Comment (wolfv) [inline include/mamba/api/configuration.hpp:28]: this is defined also a couple of lines below

Comment (wolfv) [reply include/mamba/api/configuration.hpp:28]: sorry. misread!

Comment (adriendelsalle) [reply include/mamba/api/configuration.hpp:28]: :)

Comment (adriendelsalle) [reply include/mamba/api/configuration.hpp:28]: :)

**Comment by wolfv** [comment]
looks great!

**Pull request merged**